src/main/java/de/ait/secondlife/services/CategoryServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+package de.ait.secondlife.services;
+
+import de.ait.secondlife.dto.CategoryDto;
+import de.ait.secondlife.dto.IsActiveCategoryDto;
+import de.ait.secondlife.models.Category;
+import de.ait.secondlife.repositories.CategoriesRepository;
+import de.ait.secondlife.services.interfaces.CategoryService;
+import de.ait.secondlife.services.mapping.CategoryMappingService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+
+    private final CategoriesRepository repository;
+
+    private final CategoryMappingService mappingService;
+
+    @Override
+    public CategoryDto getById(Long id) {
+
+        if (id == null || id <1){
+            throw new RuntimeException("Category ID is incorrect");
+        }
+
+        Category category = repository.findById(id).orElse(null);
+
+        if (category == null) {
+            throw new RuntimeException("Category not found");
+        }
+
+        return mappingService.mapEntityToDto(category);
+    }
+
+    @Override
+    public List<CategoryDto> getAll() {
+        return repository.findAll()
+                .stream()
+                .filter(x-> x.isActive())
+                .map(mappingService::mapEntityToDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto save(CategoryDto categoryDto) {
+
+        Category entity = mappingService.mapDtoToEntity(categoryDto);
+
+        try {
+            repository.save(entity);
+        }catch (Exception e){
+            throw new RuntimeException("Cannot save category to db");
+        }
+
+        return mappingService.mapEntityToDto(entity);
+    }
+
+    @Override
+    public CategoryDto update(Long id, CategoryDto dto) {
+
+        Category existingCategory = repository.findById(id).orElseThrow(() -> new RuntimeException("Category not found with id: " + id));
+
```

Comment:
no check whether the category is activated or not

Reply:
no need